jboss-kie-smartrouter/added/launch/jboss-kie-smartrouter.sh

```diff
@@ -75,37 +77,59 @@ function configure_router_state() {
 }
 
 function configure_router_location {
-    # DeploymentConfig environment
-    #
-    # name: KIE_SERVER_ROUTER_HOST
-    # valueFrom:
-    #   fieldRef:
-    #     fieldPath: status.podIP
-    #
-    local kieServerRouterHost="${KIE_SERVER_ROUTER_HOST}"
-    if [ "${kieServerRouterHost}" = "" ]; then
-        kieServerRouterHost="${HOSTNAME}"
-        if [ "${kieServerRouterHost}" = "" ]; then
-            kieServerRouterHost="localhost"
+
+    local routeName="${KIE_SERVER_ROUTER_NAME}"
+    local host="${KIE_SERVER_ROUTER_HOST}"
+    local port="${KIE_SERVER_ROUTER_PORT}"
+    local protocol="${KIE_SERVER_ROUTER_PROTOCOL}"
+    local routerUrlExternal="${KIE_SERVER_ROUTER_URL_EXTERNAL}"
+    local defaultInsecureHost="${HOSTNAME_HTTP:-${HOSTNAME:-localhost}}"
+    local defaultSecureHost="${HOSTNAME_HTTPS:-${defaultInsecureHost}}"
+
+    if [ "${host}" = "" ]; then
+        host="${HOSTNAME}"
+        if [ "${host}" = "" ]; then
+            host="localhost"
```

Comment:
There seems to be missing code here...

@spolti , do you know why we're not using the KIESERVER_ROUTER_SERVICE to to figure out the host and port (that is, if the HOST and PORT variables weren't explicitly defined - and thus take precedence)? Do we actually _want_ to be using the HOSTNAME and localhost here as the only fallback? I don't think so... I see KIE_SERVER_ROUTER_SERVICE in both rhpam74-prod.yaml and rhpam74-prod-immutable-monitor.yaml, and yet it's not being used in this script, and I think it should be.

For an example, you can see how we do checking and fallback, etc. here, to determine the host, port and protocol for the controller:
https://github.com/jboss-container-images/jboss-kie-modules/blob/master/jboss-kie-smartrouter/added/launch/jboss-kie-smartrouter.sh#L116-L137
Shouldn't we e doing the same thing for the smartrouter? I think so... Otherwise, why do we have all those variables being used in the template?

Reply:
@errantepiphany I don't remember why exactly we didn't use that, but I agree with you, we should use those to define the route.

Another point, Mauricio, this if can be simplified anyways:
  if [ "${host}" = "" ]; then
        host="${HOSTNAME:-localhost}"